src/webgpu/api/validation/image_copy/layout_related.spec.ts

```diff
@@ -250,68 +254,86 @@ g.test('offset_alignment')
   });
 
 g.test('bound_on_bytes_per_row')
+  .desc(`For all formats, verify image copy validations w.r.t bytesPerRow.`)
   .params(u =>
     u
       .combine('method', kImageCopyTypes)
       .combine('format', kSizedTextureFormats)
       .filter(formatCopyableWithMethod)
       .beginSubcases()
-      .combineWithParams([
-        { blocksPerRow: 2, additionalPaddingPerRow: 0, copyWidthInBlocks: 2 }, // success
-        { blocksPerRow: 2, additionalPaddingPerRow: 5, copyWidthInBlocks: 3 }, // success if bytesPerBlock <= 5
-        { blocksPerRow: 1, additionalPaddingPerRow: 0, copyWidthInBlocks: 2 }, // failure, bytesPerRow > 0
-        { blocksPerRow: 0, additionalPaddingPerRow: 0, copyWidthInBlocks: 1 }, // failure, bytesPerRow = 0
-      ])
-      .combineWithParams([
-        { copyHeightInBlocks: 0, copyDepth: 1 }, // we don't have to check the bound
-        { copyHeightInBlocks: 1, copyDepth: 0 }, // we don't have to check the bound
-        { copyHeightInBlocks: 2, copyDepth: 1 }, // we have to check the bound
-        { copyHeightInBlocks: 0, copyDepth: 2 }, // we have to check the bound
-      ])
+      .expand('params', p => {
+        const info = kTextureFormatInfo[p.format];
+        return [
+          // Copying exact fit with aligned bytesPerRow should work.
+          {
+            bytesPerRow: 256,
+            blocksPerRow: 256 / info.bytesPerBlock,
+            copyWidthInBlocks: 256 / info.bytesPerBlock,
+            success: true,
+          },
+          // Copying larger texture when padding in bytesPerRow is enough should work.
+          {
+            bytesPerRow: 256,
+            blocksPerRow: 256 / info.bytesPerBlock - 1,
+            copyWidthInBlocks: 256 / info.bytesPerBlock,
+            success: true,
+          },
+          // Unaligned bytesPerRow should not work unless the method is 'WriteTexture'.
+          {
+            bytesPerRow: 257,
```

Comment:
For alignment tests we try to test "minimally unaligned" values, i.e. 128 or 384 in this case, which is a multiple of the largest power of 2 without being a multiple of 256. That would be the most likely to erroneously pass if the alignment check was wrong. (Keeping a case for 257 also makes sense)

Reply:
Added those cases and actually removed the 257 case because it didn't make sense with the other arguments.